packages/uikit-react-native/src/components/ChannelInput/SendInput.tsx

```diff
@@ -261,6 +125,208 @@ const SendInput = forwardRef<RNTextInput, SendInputProps>(function SendInput(
+          mediaType: 'photo',
+          onOpenFailure: (error) => {
+            if (error.code === SBUError.CODE.ERR_PERMISSIONS_DENIED) {
+              alert({
+                title: STRINGS.DIALOG.ALERT_PERMISSIONS_TITLE,
+                message: STRINGS.DIALOG.ALERT_PERMISSIONS_MESSAGE(
+                  STRINGS.LABELS.PERMISSION_CAMERA,
+                  STRINGS.LABELS.PERMISSION_APP_NAME,
+                ),
+                buttons: [{ text: STRINGS.DIALOG.ALERT_PERMISSIONS_OK, onPress: () => SBUUtils.openSettings() }],
+              });
+            } else {
+              toast.show(STRINGS.TOAST.OPEN_CAMERA_ERROR, 'error');
+            }
+          },
+        });
+
+        if (mediaFile) {
+          // Image compression
+          if (
+            isImage(mediaFile.uri, mediaFile.type) &&
+            shouldCompressImage(mediaFile.type, features.imageCompressionEnabled)
+          ) {
+            await SBUUtils.safeRun(async () => {
+              const compressed = await mediaService.compressImage({
+                uri: mediaFile.uri,
+                maxWidth: imageCompressionConfig.width,
+                maxHeight: imageCompressionConfig.height,
+                compressionRate: imageCompressionConfig.compressionRate,
+              });
+
+              if (compressed) {
+                mediaFile.uri = compressed.uri;
+                mediaFile.size = compressed.size;
+              }
+            });
+          }
+
+          sendFileMessage(mediaFile);
+        }
+      },
+    });
+  }
+
+  if (input.camera.enableVideo) {
+    sheetItems.push({
+      title: STRINGS.LABELS.CHANNEL_INPUT_ATTACHMENT_CAMERA_VIDEO,
+      icon: 'camera',
+      onPress: async () => {
+        const mediaFile = await fileService.openCamera({
+          mediaType: 'video',
+          onOpenFailure: (error) => {
+            if (error.code === SBUError.CODE.ERR_PERMISSIONS_DENIED) {
+              alert({
+                title: STRINGS.DIALOG.ALERT_PERMISSIONS_TITLE,
+                message: STRINGS.DIALOG.ALERT_PERMISSIONS_MESSAGE(
+                  STRINGS.LABELS.PERMISSION_CAMERA,
+                  STRINGS.LABELS.PERMISSION_APP_NAME,
+                ),
+                buttons: [{ text: STRINGS.DIALOG.ALERT_PERMISSIONS_OK, onPress: () => SBUUtils.openSettings() }],
+              });
+            } else {
+              toast.show(STRINGS.TOAST.OPEN_CAMERA_ERROR, 'error');
+            }
+          },
+        });
+
+        if (mediaFile) {
+          sendFileMessage(mediaFile);
+        }
+      },
+    });
+  }
+
+  if (input.gallery.enablePhoto || input.gallery.enableVideo) {
+    const mediaType = (() => {
+      if (input.gallery.enablePhoto && input.gallery.enableVideo) return 'all';
+      else if (input.gallery.enablePhoto && !input.gallery.enableVideo) return 'photo';
+      else if (!input.gallery.enablePhoto && input.gallery.enableVideo) return 'video';
+      else return 'all';
```

Comment:
Small suggestion; 
If you have plan to bring [ts-pattern](https://github.com/gvergnaud/ts-pattern) <- this kind of lib some day,
```suggestion
      match(input.gallery)
        .with({ enablePhoto: true, enableVideo: false }, () => 'photo')
        .with({ enablePhoto: false, enableVideo: true }, () => 'video')
        .otherwise(() => 'all')
```
can make things more readable

Reply:
I would like to avoid exploring code from different scopes (packages) as much as possible while navigating through the code, due to it being open source. I will try modifying the code using the following approach, which is a similar pattern. thanks for the suggestion! :)

```ts
switch (true) {
  case input.gallery.enablePhoto && input.gallery.enableVideo:
    return 'all';
  case input.gallery.enablePhoto && !input.gallery.enableVideo:
    return 'photo';
  case !input.gallery.enablePhoto && input.gallery.enableVideo:
    return 'video';
  default:
    return 'all';
}
```